src/Proto3/Suite/DotProto/Generate.hs

```diff
@@ -456,45 +458,145 @@ ctxtImports tyCtxt =
 
 -- * Functions to convert 'DotProtoType' into Haskell types
 
--- | Produce the Haskell type for the given 'DotProtoType' in the
---   given 'TypeContext'
-hsTypeFromDotProto :: MonadError CompileError m => TypeContext -> DotProtoType -> m HsType
-hsTypeFromDotProto ctxt = \case
-  Prim (Named msgName)
-     | Just DotProtoKindMessage <- dotProtoTypeInfoKind <$> M.lookup msgName ctxt
-     -> HsTyApp (primType_ "Maybe") <$> hsTypeFromDotProtoPrim ctxt (Named msgName)
-  Prim pType           -> hsTypeFromDotProtoPrim ctxt pType
-  Optional (Named nm)  -> hsTypeFromDotProto ctxt (Prim (Named nm))
-  Optional pType       -> HsTyApp (primType_ "Maybe")  <$> hsTypeFromDotProtoPrim ctxt pType
-  Repeated pType       -> HsTyApp (primType_ "Vector") <$> hsTypeFromDotProtoPrim ctxt pType
-  NestedRepeated pType -> HsTyApp (primType_ "Vector") <$> hsTypeFromDotProtoPrim ctxt pType
-  Map k v              -> HsTyApp . HsTyApp (primType_ "Map")
-                          <$> hsTypeFromDotProtoPrim ctxt k
-                          <*> hsTypeFromDotProto ctxt (Prim v) -- need to 'Nest' message types
-
-hsTypeFromDotProtoPrim :: MonadError CompileError m => TypeContext -> DotProtoPrimType -> m HsType
-hsTypeFromDotProtoPrim _    Int32    = pure $ primType_ "Int32"
-hsTypeFromDotProtoPrim _    Int64    = pure $ primType_ "Int64"
-hsTypeFromDotProtoPrim _    SInt32   = pure $ primType_ "Int32"
-hsTypeFromDotProtoPrim _    SInt64   = pure $ primType_ "Int64"
-hsTypeFromDotProtoPrim _    UInt32   = pure $ primType_ "Word32"
-hsTypeFromDotProtoPrim _    UInt64   = pure $ primType_ "Word64"
-hsTypeFromDotProtoPrim _    Fixed32  = pure $ HsTyApp (protobufType_ "Fixed") (primType_ "Word32")
-hsTypeFromDotProtoPrim _    Fixed64  = pure $ HsTyApp (protobufType_ "Fixed") (primType_ "Word64")
-hsTypeFromDotProtoPrim _    SFixed32 = pure $ HsTyApp (protobufType_ "Fixed") (primType_ "Int32")
-hsTypeFromDotProtoPrim _    SFixed64 = pure $ HsTyApp (protobufType_ "Fixed") (primType_ "Int64")
-hsTypeFromDotProtoPrim _    String   = pure $ primType_ "Text"
-hsTypeFromDotProtoPrim _    Bytes    = pure $ primType_ "ByteString"
-hsTypeFromDotProtoPrim _    Bool     = pure $ primType_ "Bool"
-hsTypeFromDotProtoPrim _    Float    = pure $ primType_ "Float"
-hsTypeFromDotProtoPrim _    Double   = pure $ primType_ "Double"
-hsTypeFromDotProtoPrim ctxt (Named msgName) =
+-- Convert a dot proto type to a Haskell type
+dptToHsType :: MonadError CompileError m => TypeContext -> DotProtoType -> m HsType
+dptToHsType = foldDPT dptToHsContType dpptToHsType
+
+-- Convert a dot proto type to a wrapped Haskell type
+dptToHsTypeWrapped :: MonadError CompileError m => [DotProtoOption] -> TypeContext -> DotProtoType -> m HsType
+dptToHsTypeWrapped opts =
```

Comment:
This looks a new feature not mentioned in your summary.  Could you augment your PR summary?

Reply:
This isn't a feature but rather an implementation detail